### PR TITLE
fix: Fixed the problem that the scheduled task did not execute the compensation policy and could not get cronid from etcd

### PR DESCRIPTION
**Pull request by kakj-go**
#### What this PR does / why we need it:
Fixed the problem that the scheduled task did not execute the compensation policy and could not get cronid from etcd

#### Which issue(s) this PR fixes:

- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/all?id=283779&issueFilter__urlQuery=eyJhc3NpZ25lZUlEcyI6WyIxMDAxMjYxIl19&issueTable__urlQuery=eyJwYWdlTm8iOjEsICJwYWdlU2l6ZSI6MTB9&iterationID=1090&type=TASK)

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |       Fixed the problem that the scheduled task did not execute the compensation policy and could not get cronid from etcd       |
| 🇨🇳 中文    |       修复定时任务未执行补偿策略无法从 etcd 获取 cronid 的问题       |

### CODE REVIEW

**Comment by kakj-go** [comment]
pipeline.yml
```yaml
version: "1.1"
cron: '*/1 * * * *'
cron_compensator:
  enable: true
  latest_first: false
  stop_if_latter_executed: false
stages:
  - stage:
      - custom-script:
          alias: custom-script
          description: 运行自定义命令
          version: "1.0"
          image: registry.erda.cloud/erda-actions/custom-script-action:1.0-20211216-b1d5635
          commands:
            - sleep 130
```

test image:

Cron newly created record is not executed

image

Always follow the strategy
image

log:

image

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/pipeline/pipengine/reconciler/cron_strategy_compensator.go [hunk 1]
```diff
@@ -34,24 +34,27 @@ func (r *Reconciler) doCronCompensate(pCtx context.Context, pipelineID uint64) {
...
-		//移除cronId
-		if err := r.js.Remove(pCtx, fmt.Sprint(EtcdNeedCompensatePrefix, *pipelineWithTasks.Pipeline.CronID), nil); err != nil {
-			logrus.Infof("[doCronCompensate] can not delete etcd key, key: %s, cronId: %d, error : %v",
-				fmt.Sprint(EtcdNeedCompensatePrefix, *pipelineWithTasks.Pipeline.CronID), *pipelineWithTasks.Pipeline.CronID, err)
-		}
 
+	// 监听是否在执行的时候阻塞了补偿，阻塞了就立马补偿下, 获取etcd中当前pipeline的cron的id，然后立马删除对应的etcd的值
+	notFind, err := r.js.Notfound(pCtx, fmt.Sprint(EtcdNeedCompensatePrefix, *pipelineWithTasks.Pipeline.CronID))
+	if err != nil {
+		logrus.Infof("[doCronCompensate]: can not get cronId %d err: %v", *pipelineWithTasks.Pipeline.CronID, err)
```
Comment (sfwn) [inline modules/pipeline/pipengine/reconciler/cron_strategy_compensator.go:41]: cronID: %d, err: %v

polish all others pls

File: modules/pipeline/pipengine/reconciler/cron_strategy_compensator.go [hunk 2]
```diff
@@ -34,24 +34,27 @@ func (r *Reconciler) doCronCompensate(pCtx context.Context, pipelineID uint64) {
...
-	//监听是否在执行的时候阻塞了补偿，阻塞了就立马补偿下, 获取etcd中当前pipeline的cron的id，然后立马删除对应的etcd的值
-	if err := r.js.Get(pCtx, fmt.Sprint(EtcdNeedCompensatePrefix, *pipelineWithTasks.Pipeline.CronID), nil); err == nil {
-		//移除cronId
-		if err := r.js.Remove(pCtx, fmt.Sprint(EtcdNeedCompensatePrefix, *pipelineWithTasks.Pipeline.CronID), nil); err != nil {
-			logrus.Infof("[doCronCompensate] can not delete etcd key, key: %s, cronId: %d, error : %v",
-				fmt.Sprint(EtcdNeedCompensatePrefix, *pipelineWithTasks.Pipeline.CronID), *pipelineWithTasks.Pipeline.CronID, err)
-		}
 
+	// 监听是否在执行的时候阻塞了补偿，阻塞了就立马补偿下, 获取etcd中当前pipeline的cron的id，然后立马删除对应的etcd的值
+	notFind, err := r.js.Notfound(pCtx, fmt.Sprint(EtcdNeedCompensatePrefix, *pipelineWithTasks.Pipeline.CronID))
```
Comment (sfwn) [inline modules/pipeline/pipengine/reconciler/cron_strategy_compensator.go:39]: notFound

**Review by sfwn — CHANGES_REQUESTED** [review]

**Review by sfwn — APPROVED** [review]

**Pull request merged**